data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/PipelineConnector.java

```diff
@@ -51,6 +53,10 @@ public void enableAcknowledgements() {
         areAcknowledgementsEnabled = true;
     }
 
+    public void setReady() {
+        ready = true;
+    }
```

Comment:
I think a call to the  `start()` method should switch this connector to ready instead of having some external caller determine when this connector is ready.